lib/active_storage_validations/metadata.rb

```diff
@@ -125,7 +126,18 @@ def read_file_path
       when ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile
         file.path
       when Hash
-        File.open(file.fetch(:io)).path
+        io = file.fetch(:io)
+        if io.is_a?(StringIO)
+          tempfile = Tempfile.new([File.basename(file[:filename], '.*'), File.extname(file[:filename])])

```

Comment:
I know we have it even above in the code, but maybe we can also check if it is worth doing.

So the question is if we need to `close!` Tempfile.

Reply:
That would be nice but I don't see any nice way to do that without instance variable. Also, file created here: https://github.com/igorkasyanchuk/active_storage_validations/blob/f707239b4b486f0f1c35be49342883eb6dafb6a8/lib/active_storage_validations/metadata.rb#L68 isn't closed as well.

Reply:
Any update on this?

Reply:
According to the ruby Tempfile documentation:
> When a Tempfile object is garbage collected, or when the Ruby interpreter exits, its associated temporary file is automatically deleted. This means that’s it’s unnecessary to explicitly delete a Tempfile after use, though it’s good practice to do so
So I it should be okay to just have it garbage collected.